### PR TITLE
limit the length of request bodies

### PR DESCRIPTION
**Pull request by galdor**

### CODE REVIEW

File: agent/receiver.go [hunk 1]
```diff
@@ -70,22 +66,24 @@ func NewHTTPReceiver(conf *config.AgentConfig) *HTTPReceiver {
 		conf:        conf,
 		logger:      &errorLogger{},
 		exit:        make(chan struct{}),
+
+		maxRequestBodyLength: 10 * 1024 * 1024,
```
Comment (LeoCavaille) [inline agent/receiver.go]: Can we change that to a constant?

File: model/limited_reader.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+// LimitedReader reads from a reader up to a specific limit. When this limit
+// has been reached, any subsequent read will return
+// ErrLimitedReaderLimitReached.
+// The underlying reader has to implement io.ReadCloser so that it can be used
+// with http request bodies.
+type LimitedReader struct {
+	r     io.ReadCloser
+	limit int64
+}
```
Comment (LeoCavaille) [inline model/limited_reader.go]: https://golang.org/pkg/io/#LimitedReader :) ?

Comment (galdor) [inline model/limited_reader.go]: Read the commit message ;)

Comment (LeoCavaille) [inline model/limited_reader.go]: That's the type of things that should be in the PR description (right now there is none), when there are multiple commits I usually end up concatenating the commit messages in there (when you have a single commit it does it by default). Cool.

**Pull request merged**